schema/resource.go

```diff
@@ -125,13 +125,43 @@ func (resource *Resource) Update(updateData map[string]interface{}) error {
 	}
 	for _, property := range resource.schema.Properties {
 		id := property.ID
+
 		if val, ok := updateData[id]; ok {
-			data[id] = val
+			newVal, err := updateResourceRecursion(val, data[id])
+			if err == nil {
+				data[id] = newVal
+			} else {
+				return err
+			}
 		}
 	}
 	return nil
 }
 
+func updateResourceRecursion(updateData interface{}, sourceData interface{}) (res interface{}, err error) {
+	switch updateData.(type) {
+	case map[string]interface{}:
```

Comment:
does this code handle slices as well? in either case, please write a test for it